### DISCUSSION TITLE
역량 리포트 관련 개선안

### DISCUSSION THREAD

**Discussion by gracefulBrown** [woowacourse/prolog#601] [Ideas]
# 고민하게된 원인
역량이 변경되는 시점과 리포트가 생성되는 시점이 다르다 보니 
등록된 역량 현재 역량과 다른 상황이 발생하는거 같더라구요…

그래서 역량이 변경되는 시점과 리포트 생성시점을 똑같이 가져가면 어떨까 고민을 해보았어요.
결국 하고자 하는건 역량을 증명하기 위한 리포트를 작성하는것이라는 관점에서 생각해보니 조금(새로운) 정리가 필요하더라구요ㅠㅠ

# 고민 결과...
역량을 변경할 때 마다 이력을 남기고 역량 변경 이력을 바탕으로 리포트를 작성하면 이 문제가 해결되는것 같습니다!

# 수정되는 개념
## 역량 이력
- 역량을 수정하는 페이지에서 역량-학습로그 맵핑을 함께 하고 저장할 때 마다 역량 이력을 생성한다.
## 리포트
- 역량 이력을 바탕으로 작성한다.
- 역량에 대한 코멘트를 남길 수 있다.
- 역량에 대한 가중치를 남길 수 있다.

> 역량 이력 === 커밋
> 리포트 === 태그

# 시나리오
## 역량 최초 선택
- 기존과 동일
![image](https://user-images.githubusercontent.com/46308949/141092694-bbd1b963-2d74-4eaa-8921-c54c776cdd6c.png)

## 역량 관리 페이지
- 역량만 관리하던 페이지에서 역량 - 학습로그 맵핑 기능이 추가됨
![image](https://user-images.githubusercontent.com/46308949/141092735-c6772a57-2629-451e-b013-4be528a7069e.png)

## 역량 관리 페이지 - 학습로그 맵핑
- 수정 후 확인 버튼을 통해 역량을 저장함
- 역량을 저장할 경우 새로운 역량 이력이 생성됨
![image](https://user-images.githubusercontent.com/46308949/141092779-b8671cc7-88b2-4e84-8787-2d8e1c582beb.png)

## 역량 이력 목록 페이지
![image](https://user-images.githubusercontent.com/46308949/141092861-c6ad9e11-0551-4f59-8d2c-2b3b9a224ce9.png)

## 역량 이력 조회 페이지
- 역량 이력조회를 할 수 있음
- 리포트 생성 버튼이 있음
![image](https://user-images.githubusercontent.com/46308949/141093973-2bcbb200-3644-446b-b9fc-2f877da5fe64.png)

## 역량 관리 페이지 - 역량 이력이 있는 상황에서 새로운 이력을 남기기
- 역량의 이름을 수정할 수 있다.
- 역량을 새로 추가할 수 있다.
- 역량을 삭제할 수 있는데 삭제할 경우 맵핑되어있던 학습로그와의 맵핑이 모두 끊어진다.
![image](https://user-images.githubusercontent.com/46308949/141092779-b8671cc7-88b2-4e84-8787-2d8e1c582beb.png)

## 리포트 작성 페이지
- 역량 이력 조회 페이지에서 리포트 생성 버튼을 통해 진입한다.
- 리포트 이름과 설명을 추가한다.
- 리포트 역량의 가중치를 설정한다.
![image](https://user-images.githubusercontent.com/46308949/141094617-eebc378e-24c8-46a8-a2f6-8c96f9403ea1.png)

## 리포트 목록 페이지
![image](https://user-images.githubusercontent.com/46308949/141094794-664a72cb-a183-4f17-8cf6-66b34e078379.png)

**Comment by Hyeon9mak** [comment]
오... 진짜로 브랜치가 하나 생기는거네용

**Reply by gracefulBrown** [reply]
이력을 관리해야하는 대상이
기존에는 **역량-학습로그 맵핑정보**만 이었다면
개선안에는 **역량 자체**도 관리해야하는 대상으로 추가되는거죠!

깃의 커밋 개념이 역량 이력이 되고
깃의 태그 개념이 리포트가 될 것 같네요

**Comment by sunhpark42** [comment]
역량 관리 페이지 - 역량 이력이 있는 상황에서 새로운 이력을 남기기 이 부분이 잘이해되지 않습니다. 역량 자체에 변화가 일어나면, 기존 연동되어 있는 학습로그와의 연결은 그대로인 채로. 새로운 역량 기록이 생성된다는 걸까요???

**Reply by gracefulBrown** [reply]
네 맞습니다. 역량과 학습로그의 맵핑은 유지됩니다!
단 학습로그가 맵핑되었던 역량이 삭제될 경우에만 풀리게 됩니다.

**Comment by devhyun637** [comment]
오 브라운 역량이 바뀌거나 학습로그가 추가되면 자동으로 역량 이력이 등록되는 건 좋은 것 같아요!
다만, 제가 의문이 드는점은 `리포트의 용도`의 큰 메리트가 사라지는 것 같은데, 리포트는 어떻게 쓰일 용도일까요??

**Reply by gracefulBrown** [reply]
개선안에서 `리포트`의 역할은 `역량 이력`에 대한 보충 설명입니다.

역량 이력에는 `역량`과 `역량과 매핑된 학습로그` 정보는 있지만 역량에 대한 구체적인 설명글이 없습니다.
역량에서는 현재 역량에 대한 정보와 학습로그 맵핑에 집중을 하게하고
리포트를 통해서 현재 상태의 역량에 대한 설명과 부가 정보(가중치)를 제공하면 좋겠다 생각했어요.

모의면접에서 면접관이 볼 때 `역량`과 `역량에 맵핑된 학습로그 정보가 있지만 뭔가 부족하다는 생각이 들수 있는데,
추가적으로 역량에 대한 가중치 정보나 역량에 대한 짧은 코멘트를 통해 경험하고 학습한 부분을 언급해준다면 조금 더 돋보일 것 같다고 생각했습니다.
![image](https://user-images.githubusercontent.com/46308949/141134639-e3ea9f72-bb14-4a17-9841-e584caee5a93.png)

현재 리포트에 각 역량에 대한 추가 설명을 할 수 있는 구조가 아니지만
각 역량에 대해 짧게나마 코멘트를 넣을 수 있는 구조로 변경된다면 
리포트의 역할이 조금 더 살아나지 않을까? 생각도 해보았습니다.
